packages/components/src/styles/components/application-state.scss

```diff
@@ -7,43 +7,96 @@
 // APPLICATION-STATE COMPONENT
 //
 
-$hds-application-state-padding: 12px 0;
+$hds-application-state-content-max-width: 480px;
+$hds-application-state-footer-child-gap: 8px;
 
 .hds-application-state {
-  width: 19.5rem;
+  display: grid;
+  grid-template-columns: minmax(0, $hds-application-state-content-max-width) 1fr 1fr;
+  grid-template-rows: auto auto auto auto;
+  
   max-width: 100%;
   margin: 0 auto; // this will center the component in the parent container
+
+  &.hds-application-state--align-center {
+    grid-template-columns: 1fr minmax(0, $hds-application-state-content-max-width) 1fr;
+
+    .hds-application-state__media {
+      text-align: center;
+    }
+
+    .hds-application-state__header,
+    .hds-application-state__body,
+    .hds-application-state__footer {
+      grid-column: 2;
+    }
+  }
+}
+
+.hds-application-state__media {
+  grid-column: 1 / -1;
+  grid-row: 1;
+  
+  padding-bottom: 32px;
 }
 
 .hds-application-state__header {
+  grid-column: 1;
+  grid-row: 2;
+  
   display: grid;
   grid-template-columns: min-content 1fr;
+  grid-template-rows: auto auto;
   align-items: start;
-  color: var(--token-color-foreground-faint);
+  color: var(--token-color-foreground-strong);
+}
+
+.hds-application-state__error-code {
+  grid-column: 1 / -1;
 }
 
 .hds-application-state__icon {
+  grid-row: 2;
+  
   margin-right: 8px; // instead of column gap
   padding-top: 4px; // this seems to align the icon along with line 21
 }
 
-.hds-application-state__title,
-.hds-application-state__error-code {
-  grid-column-start: 2;
+.hds-application-state__title {
+  grid-row: 2;
+  
+  font-weight: var(--token-typography-font-weight-semibold);
+
+  &:not(.hds-application-state__icon + &) {
+    grid-column: 1 / -1;
+  }
 }
 
 .hds-application-state__body {
-  padding: $hds-application-state-padding;
-  color: var(--token-color-foreground-faint);
+  grid-column: 1;
+  grid-row: 3;
+  
+  padding: 12px 0;
```

Comment:
I know the original component uses top and bottom padding to separate the "body" from the "header" and "footer" but I would confirm if this is actually desired since the bottom padding is always present even if the component is used without a footer. (So it could possibly have been a mistake to always include bottom padding.)

Also, `margin` should normally be used (vs. `padding`) for spacing apart different items while padding is used to add inner spacing around an item's content (for example if an element has a colored background, padding can be used to prevent text inside from going to the edge of the container.)

(Margin vs. padding usage is based upon my personal understanding and philosophy so other members of the team may have differing opinions...)

Reply:
+1 on using margin (and no, margin vs padding usage in the context of a design system is not based upon personal understanding, it has become a best practice de facto)

Reply:
I was just trying to avoid stepping on anyone's toes if they disagreed for some reason...